presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -222,22 +222,30 @@ else if (nameParts.size() == 2) {
         Map<String, String> requestHeaders = getRequestHeaders(servletRequest);
         TracerHandle tracerHandle = tracerProvider.getHandleGenerator().apply(requestHeaders);
 
-        if (isTracingEnabled()) {
+        TraceTokenFactory traceTokenFactory = new PrestoTraceTokenFactory();
+
+        if (traceTokenFactory.getTraceToken(clientInfo).isPresent()) {
```

Comment:
why is TraceTokenFactory needed? Why can't we use a custom implementation of TracerProvider, which is the already supported way fo providing a custom trace token?